server/src/main/java/org/elasticsearch/search/aggregations/AggregationReduceContext.java

```diff
@@ -210,7 +215,32 @@ public boolean isFinalReduce() {
 
         @Override
         protected void consumeBucketCountAndMaybeBreak(int size) {
-            multiBucketConsumer.accept(size);
+            final String aggregationName = builder().getName();
+            try {
+                multiBucketConsumer.accept(size);
+            } catch (MultiBucketConsumerService.TooManyBucketsException e) {
+                logger.error(
+                    String.format(
+                        Locale.ROOT,
+                        "Too many buckets for aggregation [%s] (max [%d], count [%d])",
+                        aggregationName,
+                        e.getMaxBuckets(),
+                        e.getBucketsCount()
+                    )
+                );
+                throw e;
+            } catch (CircuitBreakingException e) {
+                logger.error(
```

Comment:
Also catching and re-throwing exceptions has runtime cost implications...

Reply:
I think with circuit breaker exception with change we will get two log messages in case of aggregations. Maybe we should just log where exceptions are thrown (like in `ChildMemoryCircuitBreaker` line 158 to 167)?